lib/streamlit/runtime/fragment.py

```diff
@@ -119,43 +119,109 @@ def fragment(
     *,
     run_every: int | float | timedelta | str | None = None,
 ) -> Callable[[F], F] | F:
-    """Allow a function to be run independently of the full script.
-
-    Functions decorated with ``@st.experimental_fragment`` are handled specially within
-    an app: when a widget created within an invocation of the function (a fragment) is
-    interacted with, then only that fragment is rerun rather than the full streamlit app.
+    """Decorator to turn a function into a fragment which can rerun independently\
+    of the full script.
+
+    When a user interacts with an input widget created by a fragment, Streamlit
+    only reruns the fragment instead of the full script. If ``run_every`` is set,
+    Streamlit will additionally rerun the fragment at the specified interval
+    while the session is active, even if the user is not interacting with your
+    app.
+
+    To trigger a full script rerun from inside a fragment, call ``st.rerun()``
+    directly. Any values from the fragment that need to be accessed from
+    the wider app should generally be stored in Session State.
+
+    Streamlit elements created by a fragment are cleared and redrawn on each
```

Comment:
? "created BY a fragment" -> "created INSIDE a fragment" ?

One thing that's confusing about the current language, is that I can use an outside container reference to create elements from within the fragment function ("created by the fragment") and those elements are not supposed to be cleared during a rerun. I could interpret the current language as the opposite.

Arguably, "created inside the fragment" is also a bit ambiguous, and an even more explicit way to say it would be something like "Each fragment implicitly runs in an st.container, and elements created inside that container are cleared and redrawn...". But that might be overkill.

(Also I was checking the behavior while writing this and maybe found a bug, will discuss on slack)

Reply:
Yes, I definitely need to know how the bug gets resolved because the behavior I observed is that all elements created by a fragment do get cleared and regenerated... just with some messier implications if a container (other than the sidebar) is referenced within the fragment. I didn't want to say "elements inside a fragment" since it blurs the line between the function and the elements created by the function.